### ISSUE TITLE
Not able to run jetty

### ISSUE THREAD

**Issue by schenardi** [conterra/mapapps-4-developers#12]
Trying to run jetty with the sample provided.

mvn initialize  > successful
mvn jetty:run  > ERROR

Thanks for any help

Markus

Errormessage:
[INFO] --- frontend-maven-plugin:1.12.1:gulp (run gulp ) @ mapapps-4-developers ---
[INFO] Running 'gulp.js ' in C:\data\BITBUCKET_laixo\invers_mapapps_413_UPDATE
[INFO] Error: Cannot find module 'eslint'
[INFO] Require stack:
[INFO] - C:\data\BITBUCKET_laixo\invers_mapapps_413_UPDATE\node_modules\ct-mapapps-gulp-js\dist\src\lint\gulpEsLint.js
[INFO] - C:\data\BITBUCKET_laixo\invers_mapapps_413_UPDATE\node_modules\ct-mapapps-gulp-js\dist\src\lint.js
[INFO] - C:\data\BITBUCKET_laixo\invers_mapapps_413_UPDATE\node_modules\ct-mapapps-gulp-js\dist\src\options.js
[INFO] - C:\data\BITBUCKET_laixo\invers_mapapps_413_UPDATE\node_modules\ct-mapapps-gulp-js\dist\src\index.js
[INFO] - C:\data\BITBUCKET_laixo\invers_mapapps_413_UPDATE\gulpfile.js
[INFO] - C:\data\BITBUCKET_laixo\invers_mapapps_413_UPDATE\node_modules\gulp\node_modules\gulp-cli\lib\shared\require-or-import.js
[INFO] - C:\data\BITBUCKET_laixo\invers_mapapps_413_UPDATE\node_modules\gulp\node_modules\gulp-cli\lib\versioned\^4.0.0\index.js
[INFO] - C:\data\BITBUCKET_laixo\invers_mapapps_413_UPDATE\node_modules\gulp\node_modules\gulp-cli\index.js
[INFO] - C:\data\BITBUCKET_laixo\invers_mapapps_413_UPDATE\node_modules\gulp\bin\gulp.js
[INFO]     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
[INFO]     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
[INFO]     at Module.require (node:internal/modules/cjs/loader:1005:19)
[INFO]     at require (node:internal/modules/cjs/helpers:102:18)
[INFO]     at Object.<anonymous> (C:\data\BITBUCKET_laixo\invers_mapapps_413_UPDATE\node_modules\ct-mapapps-gulp-js\dist\src\lint\gulpEsLint.js:7:18)
[INFO]     at Module._compile (node:internal/modules/cjs/loader:1103:14)
[INFO]     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
[INFO]     at Module.load (node:internal/modules/cjs/loader:981:32)
[INFO]     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
[INFO]     at Module.require (node:internal/modules/cjs/loader:1005:19) {
[INFO]   code: 'MODULE_NOT_FOUND',
[INFO]   requireStack: [
[INFO]     'C:\\data\\BITBUCKET_laixo\\invers_mapapps_413_UPDATE\\node_modules\\ct-mapapps-gulp-js\\dist\\src\\lint\\gulpEsLint.js',
[INFO]     'C:\\data\\BITBUCKET_laixo\\invers_mapapps_413_UPDATE\\node_modules\\ct-mapapps-gulp-js\\dist\\src\\lint.js',
[INFO]     'C:\\data\\BITBUCKET_laixo\\invers_mapapps_413_UPDATE\\node_modules\\ct-mapapps-gulp-js\\dist\\src\\options.js',
[INFO]     'C:\\data\\BITBUCKET_laixo\\invers_mapapps_413_UPDATE\\node_modules\\ct-mapapps-gulp-js\\dist\\src\\index.js',
[INFO]     'C:\\data\\BITBUCKET_laixo\\invers_mapapps_413_UPDATE\\gulpfile.js',
[INFO]     'C:\\data\\BITBUCKET_laixo\\invers_mapapps_413_UPDATE\\node_modules\\gulp\\node_modules\\gulp-cli\\lib\\shared\\require-or-import.js',
[INFO]     'C:\\data\\BITBUCKET_laixo\\invers_mapapps_413_UPDATE\\node_modules\\gulp\\node_modules\\gulp-cli\\lib\\versioned\\^4.0.0\\index.js',
[INFO]     'C:\\data\\BITBUCKET_laixo\\invers_mapapps_413_UPDATE\\node_modules\\gulp\\node_modules\\gulp-cli\\index.js',
[INFO]     'C:\\data\\BITBUCKET_laixo\\invers_mapapps_413_UPDATE\\node_modules\\gulp\\bin\\gulp.js'
[INFO]   ]
[INFO] }
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  24.616 s
[INFO] Finished at: 2022-05-17T11:51:27+02:00

**Comment by jessebluemr** [comment]
Can you make sure that in your pom.xml these npm/node versions are defined:
  
  <nodeVersion>v16.14.0</nodeVersion>
  <npmVersion>8.3.1</npmVersion>

The eslint dependency is auto installed as peer dependency of ct-mapapps-gulp-js.
The problem of missing "eslint" may happen if there are different versions used in by transitive dependencies.
You can declare the eslint dependency in your package.json, too. E.g. 

"eslint": "^7.32.0"

**Comment by jessebluemr** [comment]
Hm, this is strange, because I tested a fresh build without node and node_modules folder and I did not get any errors.

Did you try it with a clean mapapps-4-developers checkout?

**Comment by martinlohoff** [comment]
Sorry for late reply but I had a similar issue and resolved it like Markus.

**Comment by schenardi** [comment]
> Hm, this is strange, because I tested a fresh build without node and node_modules folder and I did not get any errors.
> 
> Did you try it with a clean mapapps-4-developers checkout?

Yes, I tried exactly the same. Could there be a dependency on the java and/or maven version?

**Comment by jessebluemr** [comment]
No. Java and maven do not interfere here.
Maven installs only the npm/node versions, which are listed in the pom.xml. 
After that `npm install` is invoked.

Maybe you have an old `package-lock.json` in your mapapps-4-developers checkout? 
If yes can you  delete it and the `node_modules` folder and try it again?

If this does not help, then we need to add the missing dependencies. Like you mentioned.
But we can still not reproduce the error :-(

**Comment by schenardi** [comment]
SOLVED!

It's a VS Code related problem > worked, as soon as Extension "Extension Pack for Java" was disabled. 
Didn't even know I enabled it once....
